### DISCUSSION TITLE
Checklist with sub options

### DISCUSSION THREAD

**Discussion by mindytom** [twilio-labs/paste#1271] [Q&A] [answered]
Hey DSYS!

I got to user test [Option 4](https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/Feedback?node-id=121%3A3600) that y'all put together. Users loved this design and found it very easy to use. Thanks team :)

Now, there is an additional sub-layer of options (schema versions) each checkbox (event type) can have. As a default, when a user is selecting an event, it will be defaulted to the latest schema version. Only when the user clicks "Show schema version" will they then see the sub-options. This is one concept I'm exploring right now. The biggest drawback is when you're looking at all the product groups (and we'll be release more soon) with all the schema versions shown, it's going to be a very very long scroll... 

Would like to see if DSYS have any recommendations on how to show and hide additional set of options, placement of it in relation to the checkboxes, and if there are any accessibility concerns with this approach. Another concept I'm considering is a modal to show the schema versions but have not designed it yet. 

Thanks!
Mindy

<img width="350" alt="Default view" src="https://user-images.githubusercontent.com/65037539/110690349-b1077480-8198-11eb-9828-75c7cda7d952.png"><img width="350" alt="Schema shown view" src="https://user-images.githubusercontent.com/65037539/110690354-b2d13800-8198-11eb-974f-66982ca7daa4.png">

--

Edit from @serifluous : Link to original thread in #1153

**Comment by jweintraub** [comment]
Love this pattern - very similar to Product Explorer page. It's a great way of filtering long pages of content into logical categories.

**Comment by TheSisb** [comment]
Hi @mindytom, sorry for the late reply, this slipped through somehow. Brought it up with the team and we will get back to you later today.

**Reply by mindytom** [reply]
No worries. 

Here's the [Invision prototype](https://twilio.invisionapp.com/share/3S10F49FQZBP#/446788702_Create_Sub) for y'all to click through the concept.

**Comment by TheSisb** [comment, marked as answer]
Hi again @mindytom, 

> Would like to see if DSYS have any recommendations on how to show and hide additional set of options, placement of it in relation to the checkboxes, and if there are any accessibility concerns with this approach. 

The vertical tab looks like a reasonable way to show and hide certain blocks of options.  

For the sub-checkboxes (v0, v1, v2), it's not really clear what those are. They likely need labeling and optionally help text. This will help with accessibility.

In terms of UI treatment, I just wanted to double check the checkbox usage, as it's unclear what it means for all v0 v1 and v2 to be selected, for example. Were there any concerns around this during design crit? 

The select all button looks enabled in one screenshot but disabled in the other. This may be confusing to users without an explanation as to why.

**Reply by corinne-nagel** [reply]
Hey @mindytom , just circling back here. Did you get the info you needed to move forward?

**Reply by mindytom** [reply]
Yes, I brought it to office hours just now and got the answers I need to move forward. Thanks Corinne for circling back to this.
